src/render/DrawContext.js

```diff
@@ -1573,5 +1582,34 @@ define([
                 " oc " + attributes.outlineColor.toHexString(true);
         };
 
+        /**
+         * A wrapper around gl.getExtension method, that also caches the extensions.
+         *
+         * @param {String} extensionName The name of the WebGL extension.
+         *
+         * @returns {Object|null} A WebGL extension object, or null if the extension is not available
+         *
+         * @throws {ArgumentError} if the name is missing or not a string
+         */
+        DrawContext.prototype.getExtension = function (extensionName) {
+            if (!extensionName) {
+                throw new ArgumentError(Logger.logMessage(Logger.LEVEL_SEVERE, "DrawContext", "getExtension",
+                    "missing extensionName"));
+            }
+
+            if (typeof extensionName !== "string") {
+                throw new ArgumentError(Logger.logMessage(Logger.LEVEL_SEVERE, "DrawContext", "getExtension",
+                    "extensionName is not a string"));
+            }
+
+            if (extensionName in this._glExtensionsCache) {
+                return this._glExtensionsCache[extensionName];
```

Comment:
This could be simplified a bit by using the opposite condition and saying: if not in the cache, try to load it. Then you always return the value from the cache. What do you think?